dali/python/nvidia/dali/types.py

```diff
@@ -252,14 +263,21 @@ def _is_scalar_shape(shape):
     return shape is None or shape == () or shape == [] or shape == 1 or \
            shape == [1] or shape == (1,)  # legacy pseudo-scalars
 
+def _is_true_scalar(value):
+    return len(getattr(value, "shape", ())) == 0
+
 def _is_mxnet_array(value):
     return 'mxnet.ndarray.ndarray.NDArray' in str(type(value))
 
 def _is_torch_tensor(value):
     return 'torch.Tensor' in str(type(value))
 
 def _is_numpy_array(value):
-    return 'numpy.ndarray' in str(type(value))
+    type_name = str(type(value))
+    return  'numpy.ndarray' in type_name or \
```

Comment:
How about, long, short, byte and other types numpy supports?

Reply:
It doesn't. It's always intN, uintN or floatN. Numpy supports array of `bool`, but the element is an ordinary Python `bool`.